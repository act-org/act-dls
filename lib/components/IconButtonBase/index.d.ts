/**
 * @prettier
 */
import * as React from 'react';
import { IconButtonProps } from '@material-ui/core';
export declare type Props = IconButtonProps;
declare const IconButtonBase: React.FC<Props>;
export default IconButtonBase;
//# sourceMappingURL=index.d.ts.map