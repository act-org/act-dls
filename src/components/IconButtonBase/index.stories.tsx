/**
 * @prettier
 */

import * as React from 'react';
import { action } from '@storybook/addon-actions';
import { Plus } from 'mdi-material-ui';

import IconButtonBase from '.';

export const base = (): React.ReactElement<any> => (
  <IconButtonBase onClick={action('onClick()')}>
    <Plus />
  </IconButtonBase>
);

export default {
  component: IconButtonBase,
  parameters: {
    info: {
      text: 'This is the base variant of the IconButton component.',
    },
  },
  title: 'Atoms/IconButton',
};
