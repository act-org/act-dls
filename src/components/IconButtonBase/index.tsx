/**
 * @prettier
 */

import * as React from 'react';
import { IconButton, IconButtonProps } from '@material-ui/core';

export type Props = IconButtonProps;

const IconButtonBase: React.FC<Props> = (
  props: Props,
): React.ReactElement<any> => <IconButton {...props} />;

export default IconButtonBase;
