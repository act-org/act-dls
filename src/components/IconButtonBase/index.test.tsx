/**
 * @prettier
 */

import * as React from 'react';
import { noop } from 'lodash';

import render from '~/helpers/test/render';

import IconButtonBase from '.';

describe('IconButtonBase', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <IconButtonBase onClick={noop}>IconButton Label</IconButtonBase>,
    );

    expect(container).toMatchSnapshot();
  });
});
