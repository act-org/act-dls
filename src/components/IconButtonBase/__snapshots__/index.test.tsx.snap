// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButtonBase matches the snapshot 1`] = `
<div>
  <button
    class="MuiButtonBase-root MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      IconButton Label
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
